pkg/apihandler/apihandler.go

```
@@ -1743,6 +1787,10 @@ func (h *SubscriptionHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 		hookData := hookBaseData(r, hookBuf.Bytes(), ctx.Variables, nil)
 		out, err := h.hooksClient.DoOperationRequest(ctx.Context, h.operation.Name, hooks.MutatingPreResolve, hookData)
 		if err != nil {
+			if ctx.Err() != nil {
```

I think it's not correct to swallow the error from the request in case of context cancellation.

---

I think it is. The client disconnected. It's a regular operation that is "propagated" through context cancellation.
The purpose of the task was to get rid of false negatives in the logs. (Error logs that should be no logs at all, as it's caused by client disconnected)

---

I think you're misunderstanding this code. if ctx.Err() is not nil, something cancelled the operation (e.g. the client, a timeout, etc..) The operation by itself did not return any error, it was the http client cancellation by the ctx.

---

I think early returning is fine in these. If we want to log client disconnects, we can do that at the top level in a single place (e.g. with middleware)

---

If the request fails, it is still an error. This is my proposal:

```go
if err != nil {
    h.log.Error("MutatingPostResolve mutation hook", abstractlogger.Error(err))
	http.Error(w, "MutatingPostResolve mutation hook failed", http.StatusInternalServerError)
   return
}
if ctx.Err() != nil {
    h.log.Debug("MutatingPostResolve mutation hook", abstractlogger.Error(ctx.Err()))
	http.Error(w, "MutatingPostResolve mutation hook failed", http.StatusInternalServerError)
    return
}
```

---

I understand the intention, but this code will never reach `if ctx.Err()` because err is never nil if ctx.Err() is != nil

---

Are you sure?. I thought someone could transport different errors through ctx and return value.